Diagnostic/DistroSpecific.py

```diff
@@ -0,0 +1,319 @@
+import time
+import subprocess
+import re
+from Utils.WAAgentUtil import waagent
+
+omi_universal_install_cmd = 'bash scx-1.6.2-337.universal.x64.sh --upgrade'
+
+
+class CommonActions:
+    def __init__(self, logger):
+        self.logger = logger
+        pass
+
+    def log_run_get_output(self, cmd):
+        """
+        Execute a command in a subshell
+        :param str cmd: The command to be executed
+        :return (int, str): A tuple of (subshell exit code, contents of stdout)
+        """
+        self.logger("RunCmd " + cmd)
+        error, msg = waagent.RunGetOutput(cmd, chk_err=True)
+        self.logger("Return " + str(error) + ":" + msg)
+        return int(error), str(msg)
+
+    def log_run_ignore_output(self, cmd):
+        """
+        Execute a command in a subshell
+        :param str cmd: The command to be executed
+        :return int: The subshell exit code
+        """
+        self.logger("RunCmd " + cmd)
+        error, msg = waagent.RunGetOutput(cmd, chk_err=True)
+        self.logger("Return " + str(error) + ":" + msg)
+        return int(error)
+
+    def log_run_with_timeout(self, cmd, timeout=3600):
+        """
+        Execute a command in a subshell, killing the subshell if it runs too long
+        :param str cmd: The command to be executed
+        :param int timeout: The maximum elapsed time, in seconds, to wait for the subshell to return; default 360
+        :return (int, str): (1, "Process timeout\n") if timeout, else (subshell exit code, contents of stdout)
+        """
+        self.logger("Run with timeout: " + cmd)
+        process = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, shell=True,
+                                   executable='/bin/bash')
+        time.sleep(1)
+        while process.poll() is None and timeout > 0:
+            time.sleep(1)
+            timeout -= 1
+        if process.poll() is None:
+            self.logger("Timeout while running:" + cmd)
+            process.kill()
+            return 1, "Process timeout\n"
+        output, error = process.communicate()
+        self.logger("Return " + str(error))
+        return int(process.returncode), output
+
+    def log_run_multiple_cmds(self, cmds, with_timeout, timeout=360):
+        """
+        Execute multiple commands in subshells, with optional timeout protection
+        :param Iterable[str] cmds: An iterable of commands to be executed
+        :param bool with_timeout: True if commands should be run with timeout
+        :param int timeout: The timeout, in seconds; default 360. Ignored if with_timeout is False.
+        :return (int, str): A tuple of (sum of status codes, concatenated stdout from commands)
+        """
+        errors = 0
+        output = []
+        for cmd in cmds:
+            if with_timeout:
+                err, msg = self.log_run_with_timeout(cmd, timeout)
+            else:
+                err, msg = self.log_run_get_output(cmd)
+            errors += err
```

Comment:
I see, and I guess then we have to do this. A good phrase I learned from this... :) Thanks.